### PR TITLE
[llvm][MachO] Fix integer truncation in rebase/bind parsing

### PR DESCRIPTION
**Pull request by zixu-w**
`Count` and `Skip` should use `uint64_t` as they are encoded/decoded
using 64-bit ULEB128.

In `*_OPCODE_DO_*_ULEB_TIMES_SKIPPING_ULEB`, `Skip` could be encoded as
a two's complement for moving `SegmentOffset` backwards. Having a 32-bit
`Skip` truncates the encoded value and leads to a malformed `AdvanceAmount`
and invalid `SegmentOffset` that extends past valid sections.

### CODE REVIEW

**Comment by zixu-w** [comment]
Trying to figure out how to construct a minimum reproducer code/binary/object YAML for a test case.

File: llvm/test/Object/macho-bind-negative-skip.test [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// A valid MachO object with a bind table containing an opcode
+// `BIND_OPCODE_DO_BIND_ULEB_TIMES_SKIPPING_ULEB` with negative skip value
+// (0xFFFFFFFFFFFFFFF0).
+// libObject should be able to correctly parse the bind table without crashing.
```
Comment (cyndyishida) [inline llvm/test/Object/macho-bind-negative-skip.test:4]: I think this comment is already implied by llvm-objdump's exit status.

File: llvm/test/Object/macho-bind-negative-skip.test [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+
+RUN: yaml2obj %p/Inputs/MachO/bind-negative-skip.yaml | llvm-objdump --bind --macho -
```
Comment (cyndyishida) [inline llvm/test/Object/macho-bind-negative-skip.test:6]: Not crashing is one part, whats the expected output?

**Review by cyndyishida — APPROVED** [review]
LGTM

**Pull request merged**